packages/cli/src/commands/create2.ts

```diff
@@ -2,48 +2,64 @@ import pickBy from 'lodash.pickby';
 
 import create from '../scripts/create';
 import queryDeployment from '../scripts/query-deployment';
+import querySignedDeployment from '../scripts/query-signed-deployment';
 import { parseInit } from '../utils/input';
 import { fromContractFullName } from '../utils/naming';
 import { hasToMigrateProject } from '../utils/prompt-migration';
 import ConfigVariablesInitializer from '../models/initializer/ConfigVariablesInitializer';
 
 const name: string = 'create2';
 const signature: string = `${name} [alias]`;
-const description: string = 'deploys a new upgradeable contract instance using CREATE2 at a predetermined address given a numeric <salt> and a <from> address. Provide the <alias> you added your contract with, or <package>/<alias> to create a contract from a linked package. Warning: support for this feature is experimental.';
+const description: string = 'deploys a new upgradeable contract instance using CREATE2 at a predetermined address given a numeric <salt> and a <from> address. Provide the <alias> you added your contract with, or <package>/<alias> to create a contract from a linked package. A <signature> can be provided to derive the deployment address from a signer different to the <from> address. Warning: support for this feature is experimental.';
 
 const register: (program: any) => any = (program) => program
   .command(signature, undefined, { noHelp: true })
   .usage('[alias] --network <network> --salt <salt> [options]')
   .description(description)
   .option('--salt <salt>', `salt used to determine the deployment address`)
-  .option('--query', `do not create the contract and just return the deployment address`)
-  .option('--init [function]', `call function after creating contract. If none is given, 'initialize' will be used`)
-  .option('--args <arg1, arg2, ...>', 'provide initialization arguments for your contract if required')
+  .option('--query [sender]', `do not create the contract and just return the deployment address, optionally specifying the sender used to derive the deployment address (defaults to 'from')`)
+  .option('--init [function]', `optional initialization function to call after creating contract (defaults to 'initialize')`)
+  .option('--args <arg1, arg2, ...>', 'optional arguments to the initialization function')
+  .option('--admin <admin>', 'optional admin of the proxy (uses the project\'s proxy admin if not specified)')
+  .option('--signature <signature>', `optional signature of the request, uses the signer instead of sender to derive the deployment address`)
   .option('--force', 'force creation even if contracts have local modifications')
   .withNetworkOptions()
   .action(action);
 
 async function action(contractFullName: string, options: any): Promise<void> {
   const { network, txParams } = await ConfigVariablesInitializer.initNetworkConfiguration(options);
   if (!await hasToMigrateProject(network)) process.exit(0);
+  if (!options.salt) throw new Error('option `--salt\' is required');
```

Comment:
Using ` and \' is a convention I don't know, or is it a typo?

Reply:
It's how commander.js displays errors